common/sgx_workload/workload/ext_work_order_info_kme.cpp

```diff
@@ -281,6 +281,14 @@ int ExtWorkOrderInfoKME::CreateWorkOrderKeyInfo(
         int count = this->in_work_order_keys.size();
         wo_key_info.in_data_keys.resize(count);
         for (int i=0; i<count; i++) {
+            std::string in_data_enc_key = \
+                ByteArrayToStr(wo_key_info.in_data_keys[i].decrypted_data);
+            // skip encrypting input data encryption key if
+            // its value is equivalent to empty or null or "-"
+            if (in_data_enc_key.empty() || "null" == in_data_enc_key ||
```

Comment:
According to spec
If this key is not provided or set to "null" or to "", the data is encrypted using encryptedSessionKey from the Work Order request

If the key value is set to "-", the data item is not encrypted, a.k.a. sent as clear text

https://entethalliance.github.io/trusted-computing/spec.html#sec-work-order-input-and-output-data-format

Reply:
That's correct from Singleton perspective. At KME, while preprocessing, if dataEncryptionKey is empty or null or "-", i have to skip encrypting the data encryption key by using one time symmetric key that gets generated during preprocess. My comment is to explaining this.

Reply:
From KME side all("null", "", "-") are treated as same? Who is handling "null" and "" to de-crypt data using encryptedSessionKey? Is WPE doing that?

Reply:
Yes, they all treated the same way. all these checks are performed again at WPE while doing actual work order processing..

Reply:
Please add that comment here because it slightly different from spec of singleton worker.

Reply:
Done